### DISCUSSION TITLE
When I use two backgroundPlotters, I get error once close the window

### DISCUSSION THREAD

**Discussion by HametAl** [pyvista/pyvista#4447] [Q&A]
Hi
I want to use two backgroundplotters: 1 plotter in the main window and 1 plotter in the docked window. When I close the application, it goes to a loop and gives errors. If  I close the docked plotter and then close the main window it doesn't give any error. I used sys.exit(app.exec_()), but didn't work. I also tried to close the docked plotter with atexit() again didn't work. Is there any way to close all plotters, when closing the application using x button on the top right corner of the main window?

Thanks

**Comment by banesullivan** [comment]
cc @pyvista/pyvistaqt-admin

**Comment by larsoner** [comment]
I don't think I've ever embedded two BackgroundPlotter instances in the same window, it's possible there is some bug with how it works. Can you make a minimal example to show the problem?

**Reply by HametAl** [reply]
I have attached a screenshot of the error and
![error](https://github.com/pyvista/pyvista/assets/125446969/9ddeaeeb-cc55-4af8-b877-bd272a5c797b)
I have summarized my code here:

from pyvistaqt import BackgroundPlotter
from pyvistaqt import QtInteractor
......
plotter = BackgroundPlotter(menu_bar=False,toolbar=False)
sectionPlotter = QtInteractor()
.....
dock = QDockWidget('Properties Panel')
dock.setFeatures(QDockWidget.DockWidgetFloatable|QDockWidget.DockWidgetMovable)
dock.setMinimumWidth(370)
dockwidget=QWidget()
plotter.app_window.addDockWidget(QtCore.Qt.LeftDockWidgetArea,dock)
toolbox = QToolBox()
dock.setWidget(toolbox)
.....
lineProperties = QWidget()
lineProperties_layout=QVBoxLayout()
lineProperties.setLayout(lineProperties_layout)
toolbox.addItem(lineProperties, 'Physical Element')
.....
elementTypeGroup = QGroupBox('Element Type')
elementTypeGroup_layout = QGridLayout()
elementTypeGroup.setLayout(elementTypeGroup_layout)
lineProperties_layout.addWidget(elementTypeGroup)
sectionProperties = QWidget()
sectionProperties_layout=QVBoxLayout()
sectionProperties.setLayout(sectionProperties_layout)
toolbox.addItem(sectionProperties, 'Section Designer')
.....
sectionDesignerGroup = QGroupBox('Section properties')
sectionDesignerGroup_layout = QGridLayout()
sectionDesignerGroup.setLayout(sectionDesignerGroup_layout)
sectionProperties_layout.addWidget(sectionDesignerGroup)

......
sectionDesignerGroup_layout.addWidget(sectionPlotter, 0, 0, 1, 1)

**Reply by HametAl** [reply]
I read your Qtinteractor help again, and I found the solution. I just added the following lines and it worked:

def closeSectionPlotter():
    sectionPlotter.close()
    plotter.close() 

plotter.app_window.signal_close.connect(closeSectionPlotter)

Thank you very much for your great code.
